be/src/util/cpu_info.cpp

```diff
@@ -227,6 +241,76 @@ void CpuInfo::_init_numa() {
     _init_numa_node_to_cores();
 }
 
+void CpuInfo::_init_num_cores_with_cgroup() {
+    bool running_in_docker = fs::path_exist("/.dockerenv");
+    if (!running_in_docker) {
+        return;
+    }
+    struct statfs fs;
+    if (statfs("/sys/fs/cgroup", &fs) < 0) {
+        LOG(WARNING) << "Fail to get file system statistics. err: " << errno_to_string(errno);
+        return;
+    }
+    std::ifstream ifs;
+    int64_t cfs_period_us = -1;
+    int64_t cfs_quota_us = -1;
+    if (fs.f_type == TMPFS_MAGIC) {
+        // cgroup v1
+        ifs.open("/sys/fs/cgroup/cpu/cpu.cfs_period_us");
```

Comment:
If the docker container runs with the specific `cpuset-cpus`, should we consider it by [sched_getaffinity](https://stackoverflow.com/questions/10490756/how-to-use-sched-getaffinity-and-sched-setaffinity-in-linux-from-c/50117787#50117787), or does the CPU bandwidth control already considers `cpuset-cpus`?
``` bash
sudo docker run --cpuset-cpus="0-10" ...
```

Reply:
updated